app/Actions/PhotoAuthorisationProvider.php

```diff
@@ -67,21 +75,52 @@ public function applyVisibilityFilter(Builder $query): Builder
 	 * See {@link PhotoAuthorisationProvider::applyVisibilityFilter()} for a
 	 * specification of the rules when a photo is visible.
 	 *
-	 * @param string $photoID
+	 * @param Photo|null $photo the photo; `null` is accepted for convenience
+	 *                          and the `null` photo is always authorized
+	 *
+	 * @return bool true, if the authenticated user is authorized
+	 */
+	public function isVisible(?Photo $photo): bool
+	{
+		return
+			$photo === null ||
+			AccessControl::is_current_user($photo->owner_id) ||
+			$photo->is_public ||
```

Comment:
To check, but does this property is the computed result of album & photo public or is it the value in the DB ? :thinking:

Reply:
It is the computed result.